### PR TITLE
Adjust duration calculation to account for precision

### PR DESCRIPTION
**Pull request by julien-gadonneix**
## What does this PR do?

Small fix for decimal precision in `EnsureTexts` `EventsTransform` to be sure that all the `Word` events are enclosed in the newly created `Text` events.
Reused the same approach as in `_extract_sentences` in `utils`

### CODE REVIEW

**Comment by kingjr** [comment]
I'm not entirely sure this is the right place for the fix, I'll defer to @jrapin

**Comment by kingjr** [comment]
can you add a unit test?

**Comment by kingjr** [comment]
@jrapin ok for you?

**Comment by bkowshik** [comment]
Following up on @kingjr's question about whether this is the right place for the fix — I think it might be worth handling it one layer down. The root cause is a comparison issue rather than a data one. 

One side effect: there are now two workarounds for the same issue — this PR pads `duration` by `+ 2*eps`, while `_extract_sentences` in `utils.py` pads both edges (`start - eps` and `+ 2*eps`).

An alternative would be a single tolerance in `find_enclosed` (only two real callers):

```python
def find_enclosed(df, start, duration, *, tolerance: float = 1e-6):
    estart = np.asarray(df.start, dtype=float)
    estop = estart + np.asarray(df.duration, dtype=float)
    is_enclosed = (estart >= start - tolerance) & (estop <= start + duration + tolerance)
    return pd.Series(df.index[is_enclosed])
```

`EnsureTexts` could then revert to `duration = stop - start` (and the `eps` in `_extract_sentences` could go too) — same `1e-6`, just moved to where the rounding is compared, so durations stay accurate and the edge case is handled in one place. The existing test passes as-is since it already asserts via `find_enclosed`.

Either approach fixes the immediate bug — flagging the option in case it's useful.

**Comment by kingjr** [comment]
definitely better to add tolerance to find_enclosed.

But I **think** (not sure) it should be 0 by default

**Review by kingjr — DISMISSED** [review]
add tolerance to find_enclosed.

File: neuralset-repo/neuralset/events/transforms/splitting.py [hunk 1]
```diff
@@ -157,6 +157,7 @@ class AssignWordSplitAndContext(EventsTransform):
     ratios: tuple[float, float, float] = (0.8, 0.1, 0.1)
     seed: int = 0
     max_unmatched_ratio: float = 0.0
+    tol: float = 0.0
```
Comment (kingjr) [inline neuralset-repo/neuralset/events/transforms/splitting.py]: tolerance

File: neuralset-repo/neuralset/events/transforms/text.py [hunk 2]
```diff
@@ -133,11 +133,14 @@ class AddSentenceToWords(EventsTransform):
     ----------
     max_unmatched_ratio : float
         Maximum ratio of words without a character-positioned match.
+    tol : float, optional
+        Tolerance for Text event start and end times. Default is 0.0.
```
Comment (kingjr) [inline neuralset-repo/neuralset/events/transforms/text.py]: 0. sec.

**Comment by kingjr** [comment]
thanks @julien-gadonneix !
